test/functional/web/safari-nativewebtap-e2e-specs.js

```diff
@@ -1,270 +1,268 @@
-//         });
-
-//         describe('with tabs -', function () {
-//           before(async function () {
-//             if (skipped || !deviceName.toLowerCase().includes('ipad')) {
-//               return this.skip();
-//             }
-//             await loadPage(driver, GUINEA_PIG_PAGE);
-
-//             // open a new tab and go to it
-//             await driver.elementByLinkText('i am a new window link').click();
-
-//             await retryInterval(10, 1000, async function () {
-//               await driver.title().should.eventually.eql('I am another page title');
-//             });
-//           });
-
-//           it('should be able to tap on an element', async function () {
-//             await loadPage(driver, GUINEA_PIG_PAGE);
-
-//             await driver.elementByLinkText(PAGE_3_LINK).click();
-
-//             await spinTitleEquals(driver, PAGE_3_TITLE, SPIN_RETRIES);
-
-//             await driver.back();
-
-//             // try again, just to make sure
-//             await driver.elementByLinkText(PAGE_3_LINK).click();
+import chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import _ from 'lodash';
+import { initSession, deleteSession, hasDefaultPrebuiltWDA, MOCHA_TIMEOUT } from '../helpers/session';
+import { amendCapabilities, SAFARI_CAPS, DEVICE_NAME } from '../desired';
+import {
+  openPage, spinTitleEquals, spinTitle, GUINEA_PIG_PAGE,
+  GUINEA_PIG_SCROLLABLE_PAGE, GUINEA_PIG_APP_BANNER_PAGE
+} from './helpers';
+import { retryInterval } from 'asyncbox';
+import B from 'bluebird';
+import Simctl from 'node-simctl';
+import { performance } from 'perf_hooks';
+
+/**
+ * This test suite can be affected by two environment variables:
+ *   1. ALL_DEVICES - will use simctl to get _all_ the iPhone and iPad device
+ *                    types available on the current Xcode installation, and
+ *                    runs the tests for each one. This is a long process.
+ *   2. DEVICE_NAME - the name of a particular device. The tests will be run
+ *                    against that device only
+ * If neither of these are provided, the tests will be run against a subset
+ * of devices that are available (one iPad, on regular iPhone, and the bigger
+ * new iPhones that have proven problematic in the past)
+ */
+
+
+chai.should();
+chai.use(chaiAsPromised);
+
+const caps = amendCapabilities(SAFARI_CAPS, {
+  'appium:safariInitialUrl': GUINEA_PIG_PAGE,
+  'appium:nativeWebTap': true,
+  'appium:usePrebuiltWDA': hasDefaultPrebuiltWDA(),
+});
+
+const SPIN_RETRIES = 25;
+
+const PAGE_3_LINK = 'i am a link to page 3';
+const PAGE_3_TITLE = 'Another Page: page 3';
+
+describe('Safari - coordinate conversion -', function () {
+  this.timeout(MOCHA_TIMEOUT * 2);
+
+  let devices = [];
+  before(async function () {
+    if (process.env.REAL_DEVICE) {
+      // skip, by not having any devices in the list
+    } else if (process.env.ALL_DEVICES) {
```

Comment:
I'm not sure if we should keep running this tests with`ALL_DEVICES=1` on CI as it seems to end up with timeout error.
ref:
- https://github.com/appium/appium-xcuitest-driver/blob/master/ci-jobs/templates/xcuitest-e2e-template.yml#L76
- https://github.com/appium/appium-xcuitest-driver/blob/master/package.json#L113

Reply:
I would say it is also some legacy stuff. Feel free to get rid of it if you see it's not used anywhere else

Reply:
thank you. I've removed legacy `ALL_DEVICES` flag as https://github.com/appium/appium-xcuitest-driver/pull/1490/commits/ec40db43fe88075629b9f737bef3b8c8a5648251